File: packintel_ai/backend/app/schemas/compare.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from app.schemas.material import MaterialDetail
from app.schemas.recommendation import RecommendRequest


class CompareRequest(BaseModel):
    material_ids: List[str] = Field(..., min_length=2, max_length=5)
    food_commodity: Optional[str] = None
    food_properties: Optional[RecommendRequest] = None
    demo_mode: Optional[bool] = None


class PropertyMatrix(BaseModel):
    properties: List[str]
    data: Dict[str, Dict[str, float]]


class RadarDataPoint(BaseModel):
    property: str
    values: Dict[str, float]


class CompareResponse(BaseModel):
    comparison_id: str
    materials: List[MaterialDetail]
    comparison_matrix: PropertyMatrix
    radar_data: List[RadarDataPoint]
    winner_by_property: Dict[str, str]
    overall_ranking: List[dict]

File: packintel_ai/frontend/src/api/compare.ts
import apiClient from './client'

export interface CompareRequest {
  material_ids: string[]
  food_commodity?: string
  demo_mode?: boolean
}

export interface RadarDataPoint {
  property: string
  values: Record<string, number>
  raw_values: Record<string, number>
  unit: string
}

export type RadarChartData = RadarDataPoint

export interface CompareResponse {
  comparison_id: string
  materials: import('@/types/materials').MaterialDetail[]
  comparison_matrix: {
    properties: string[]
    data: Record<string, Record<string, number>>
  }
  radar_data: RadarDataPoint[]
  winner_by_property: Record<string, string>
  overall_ranking: Array<{
    rank: number
    material_name: string
    overall_score: number
  }>
}

export const compareApi = {
  compare: async (payload: CompareRequest): Promise<CompareResponse> => {
    const response = await apiClient.post('/api/v1/compare', payload)
    return response.data
  },
}
